cpp/include/cuspatial/cubic_spline.hpp

```diff
@@ -84,4 +84,36 @@ std::unique_ptr<cudf::table> cubicspline_coefficients(
   cudf::column_view const& ids,
   cudf::column_view const& offsets,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+namespace detail {
```

Comment:
Does the detail namespace need to be exposed in the public header here?

Reply:
No, I will create a `detail` folder and move this definition there.

Reply:
If it's not used by other compilation units, it may not be needed outside of the .cpp/.cu files at all.